pccommon/pccommon/middleware.py

```diff
@@ -1,96 +1,99 @@
 import asyncio
 import logging
 import time
-from typing import Awaitable, Callable
+from functools import wraps
+from typing import Any, Callable
 
-from fastapi import HTTPException, Request, Response
+from fastapi import HTTPException, Request
 from fastapi.applications import FastAPI
-from starlette.middleware.base import BaseHTTPMiddleware
-from starlette.responses import PlainTextResponse
+from fastapi.dependencies.utils import (
+    get_body_field,
+    get_dependant,
+    get_parameterless_sub_dependant,
+)
+from fastapi.responses import PlainTextResponse
+from fastapi.routing import APIRoute, request_response
 from starlette.status import HTTP_504_GATEWAY_TIMEOUT
-from starlette.types import Message
 
 from pccommon.logging import get_custom_dimensions
-from pccommon.tracing import trace_request
 
 logger = logging.getLogger(__name__)
 
 
-async def handle_exceptions(
-    request: Request,
-    call_next: Callable[[Request], Awaitable[Response]],
-) -> Response:
-    try:
-        return await call_next(request)
-    except HTTPException:
+async def http_exception_handler(request: Request, exc: Exception) -> Any:
+    # Log the exception with additional request info if needed
+    logger.exception("Exception when handling request", exc_info=exc)
+    # Return a custom response for HTTPException
+    if isinstance(exc, HTTPException):
         raise
-    except Exception as e:
+    # Handle other exceptions, possibly with a generic response
+    else:
         logger.exception(
             "Exception when handling request",
-            extra=get_custom_dimensions({"stackTrace": f"{e}"}, request),
+            extra=get_custom_dimensions({"stackTrace": f"{exc}"}, request),
         )
         raise
 
 
-class RequestTracingMiddleware(BaseHTTPMiddleware):
```

Comment:
It looks like we lost the tracing functionality we had previously. This middleware is running [some code to enhance the metadata](https://github.com/microsoft/planetary-computer-apis/blob/2c23410ce56fbcba751df4e732422539545e9959/pccommon/pccommon/tracing.py#L41) on the request which we use to generate metrics and reports for collections and service usage. We'll have to figure out a way to preserve that in this project.

I'd have to revisit how we managed this in Spatio, but I think the pressure there was reduced because we use internal tools for these metrics.